cmd/debugger/main.go

```diff
@@ -164,9 +165,30 @@ func interactiveMode(ctx context.Context, remoteConsoleAddr string, cmdBuilder f
 		return err
 	}
 
+	// once the command completes, waitErrSet will be set to true, indicating the command finished (or failed)
+	// if is is not set, then that means the interactive debugger has exited before the wrapped command has finished.
+	var waitErrMu sync.Mutex
+	var waitErr error
+	var waitErrSet bool
+
+	hasCommandFinished := func() bool {
+		// give c.Wait() time to acquire the lock first, to detect if the command closed as expected
+		time.Sleep(time.Millisecond * 10)
+		waitErrMu.Lock()
+		defer waitErrMu.Unlock()
+		return waitErrSet
+	}
+
+	logErrorIfNonCleanExit := func(console conslogging.ConsoleLogger, err error) {
+		if hasCommandFinished() {
+			return
+		}
+		console.VerboseWarnf(err.Error())
+	}
+
 	ptmx, err := pty.Start(c)
 	if err != nil {
-		log.Error(errors.Wrap(err, "failed to start pty"))
+		conslogger.VerboseWarnf("failed to start pty\n")
```

Comment:
if this fails, everything fails, it should stay as an error.

Reply:
i agree, my thinking was that showing this as a warning (always) might confuse users with the build error they are experiencing and diagnosing a problem to use the interactive debugger can be further investigated in verbose mode.

on the other end we do have now `earthly debugger` as the prefix, so I don't feel strongly about the above.